pkg/cmd/pr/create/create.go

```diff
@@ -731,10 +739,11 @@ func handlePush(opts CreateOptions, ctx CreateContext) error {
 			for {
 				r := NewRegexpWriter(opts.IO.ErrOut, gitPushRegexp, "")
 				defer r.Flush()
-				cmdErr := r
-				cmdIn := opts.IO.In
-				cmdOut := opts.IO.Out
-				if err := git.Push(headRemote.Name, fmt.Sprintf("HEAD:%s", ctx.HeadBranch), cmdIn, cmdOut, cmdErr); err != nil {
+				gitClient := ctx.GitClient
+				oldStderr := gitClient.Stderr
+				gitClient.Stderr = r
+				defer func() { gitClient.Stderr = oldStderr }()
```

Comment:
It is kind of annoying to have to reset the stderr for the client but didn't see another way to do this without introducing a new function on the client that took in override options such as `stderr`.

Reply:
This is why I had suggested https://github.com/cli/go-gh/pull/75#discussion_r976466084

Perhaps those commands that need overrides could accept optional extra arguments. Then, for example:
```go
gitClient.Push(ctx, remoteName, headBranch, git.WithStderr(w))
```

Reply:
Added `CommandModifier` type to address this.